lkcrypto.c

```diff
@@ -549,6 +549,7 @@ static int init(lua_State *L, struct hash_lock_t *state, struct list_head *list,
 	size_t name_len = 0;
 
 	if (get_status(state) != INIT) {
+		complete(task);
```

Comment:
I'm not sure about this one, it seems that we can only be here after first hitting the `set_status(state, IN_PROGRESS)` below so `complete(task)` has already been called. Is it safe to call it more than once on the same task? I think this call is unnecessary in any case and wasteful at best, but if we're not sure of its safety it should definitely not be here.

Reply:
I think I'll remove this one. What about the lower one? Do I need to check if it's been completed?

Reply:
Do the docs say anything? https://www.kernel.org/doc/html/latest/scheduler/completion.html

Reply:
Ok, sounds good then.